src/app/feed/shared/feed-connection.model.spec.ts

```diff
@@ -30,102 +30,248 @@ describe("Service: FeedConnection", () => {
+    });
 
-    expect(this.pluginHandle.handleRemoteJsep).toHaveBeenCalledWith(
-      jasmine.objectContaining({
-        jsep: jsep
-      })
-    );
   });
 
-  it("should call pluginHandle.data when call sendData", () => {
-    let fConnection: FeedConnection = new FeedConnection();
-    fConnection.setAttrs(
-      this.pluginHandle,
-      1
-    );
+  describe("#subscribe", () => {
+    beforeEach(() => {
+      this.jsepVal = {};
+    });
+    it("should create a webRTC answer", () => {
+      this.connection.subscribe(this.jsepVal);
+
+      expect(this.pluginHandle.createAnswer).toHaveBeenCalledWith(
+        jasmine.objectContaining({
+          jsep: this.jsepVal,
+          media: {
+            audioSend: false,
+            videoSend: false,
+            data: true
+          },
+          success: jasmine.any(Function),
+          error: jasmine.any(Function)
+        })
+      );
+    });
+
+    it("should send a request start message when webRTC answer success", () => {
+      this.pluginHandle.createAnswer.and.callFake((options) => {
+        options.success(this.jsepVal);
+      });
+      this.connection.subscribe(this.jsepVal);
+
+      expect(this.pluginHandle.send).toHaveBeenCalledWith(
+        jasmine.objectContaining({
+          message: {
+            request: "start",
+            room: 1
+          },
+          jsep: this.jsepVal
+        })
+      );
+    });
+
+    it("should shows an error when webRTC answer fails", () => {
+      this.pluginHandle.createAnswer.and.callFake((options) => {
+        options.error();
+      });
+      spyOn(window.console, "error");
+      this.connection.subscribe(this.jsepVal);
+
+      expect(window.console.error).toHaveBeenCalled();
+    });
+  });
+
+  describe("#setConfig", () => {
+    beforeEach(() => {
+      this.options = {
+        values: {
+          audio: true,
+          video: true
+        },
+        ok: jasmine.createSpy("okCallback")
+      };
+    });
+
+    it("should create a new config if not exist", () => {
+      expect(this.connection.config).toBe(null);
+      this.connection.setConfig(this.options);
 
-    let data = { id: 1 };
-    fConnection.sendData(data);
+      expect(this.connection.config).not.toBe(null);
+    });
+
+    it("should update the existing config", () => {
+      this.connection.config = jasmine.createSpyObj("config", ["set"]);
+      this.connection.setConfig(this.options);
+
+      expect(this.connection.config.set).toHaveBeenCalledWith(this.options);
+    });
+  });
+
+  describe("#getConfig", () => {
+    it("should return undefined if config not exist", () => {
+      let result: any = this.connection.getConfig();
+      expect(result).not.toBeDefined();
+    });
+
+    it("should return the existing config", () => {
+      this.connection.config = jasmine.createSpyObj("config", ["get"]);
+      let result: any = this.connection.getConfig();
+
+      expect(this.connection.config.get).toHaveBeenCalled();
```

Comment:
You're testing the `config.get` is called but you're not testing whether it's returned. So the following implementation would pass the test:

```javascript
if (this.config) {
  this.config.get();
  return "another-thing";
}
return;
```